File: LeetCode/771/771.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> st(jewels.begin(), jewels.end());
        return std::ranges::count_if(stones, [&st](const auto& chr){return st.contains(chr);});
    }
};

File: LeetCode/771/771.rs
use std::collections::HashSet;
impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        let hash: HashSet<char> = jewels.chars().collect();
        stones.chars().filter(|x| hash.contains(&x)).count() as i32
        
    }
}
